docs/Decentraland Protocol for Explorers.md

```diff
@@ -0,0 +1,135 @@
+
+### Content layer
+
+The Decentraland Explorer and other applications render representations of the assets from the consensus layer. The content layer defines the mechanisms in which the content is:
+
+1. **Signed**, **validated** and **uploaded** to a node
+2. **Replicated** to other nodes and
+3. **Served** to the final users
+
+To do so, a [content-server](https://github.com/decentraland/catalyst) was created. It is a service that runs inside a Catalyst.
+
+> A _catalyst node_ is a bundle of services that self-contain a copy of Decentraland. The DAO [governs a curated list of trusted catalysts](https://governance.decentraland.org/?type=catalyst) that hold a working copy of Decentraland. It contains everything necessary to make the Explorer work.
+
+The content server stores and synchronizes _Entities_.
+
+An [Entity](https://github.com/decentraland/common-schemas/blob/be7213b40a2180a9a99035eb87e8a5d4b8438e7f/src/platform/entity.ts#L21-L37) is a signed data structure holding a list of content files, a deployment date and pointers. A pointer is a human readable "shortcut" to the entity i.e. the [`0,0`](https://peer.decentraland.org/content/entities/scenes?pointer=0,0) representation of a LAND. Every time an entity is deployed and accepted by the network, the pointers for that entity will now point to the newest entity for that set of pointers. That is the mechanism used to change the content of the land, wearables and emotes.
+
+Unlike LAND, wearables and emotes have [URN pointers](https://github.com/decentraland/urn-resolver/blob/b11aeb677e06e1a9e1d7994efa98a5f11867f854/test/urn.spec.ts#L138-L147). URN are used to reference any asset inside Decentraland, the technology was selected in pursue of a common identifier that enables [extensibility to other networks or remote assets](https://github.com/decentraland/urn-resolver/blob/b11aeb677e06e1a9e1d7994efa98a5f11867f854/test/urn.spec.ts#L269-L280) and [interoperability](https://github.com/common-metaverse/urn-namespaces) with other platforms.
+
+> TODO: Describe entities format
+
+> TODO: Describe entities content restrictions
+
+> TODO: Describe entities IPFS CIDs
+
+> TODO: Describe AuthChain
+
+#### Endpoints that are part of the protocol
+
+##### `GET /ipfs/:cid` Download an entity or content file
+
+Used to download the content. This endpoint is used by the content-server synchronization and by the explorer.
```

Comment:
```suggestion

```

This endpoint is not part of the protocol, it's an endpoint only served at peer-historical.decentraland.org

Reply:
I'd like to propose this endpoint as an alias for `/contents/:cid`

The rationale is that it is the only endpoint used to download content once you know a deployed entity CID.

And having the `/ipfs/` prefix instead of `/contents/` will enable us to point directly to IPFS to load a specific scene.

Here is a working example: https://play.decentraland.org/?GLOBAL_PX=urn:decentraland:entity:bafkreiciy2hhxyz6vqiqsqy2z2vzqibdow6tf2bleg2pk7frxrnnijy4zi?baseUrl=https://sdk-content-server.decentraland.org/ipfs/

It loads the portable experience `urn:decentraland:entity:bafkreiciy2hhxyz6vqiqsqy2z2vzqibdow6tf2bleg2pk7frxrnnijy4zi` from `?baseUrl=https://sdk-content-server.decentraland.org/ipfs/`

Reply:
okay, but wouldn't it be better to avoid exposing the tech implementation and expose it as `/contents` to be redundant with the layers proposed?

Reply:
roger!